File: tests/test_chatbot_stability.py
#!/usr/bin/env python3
"""
Test the chatbot API without external process interference
"""
import requests
import time
import json
from datetime import datetime

def test_chatbot():
    server_url = 'http://localhost:3000'
    
    print('\n' + '='*60)
    print('🤖 CHATBOT STABILITY TEST (Python)')
    print('='*60 + '\n')
    
    test_messages = [
        'Hello! How are you?',
        'How do I keep my kidneys healthy?',
        'What is a kidney stone?',
    ]
    
    for idx, message in enumerate(test_messages, 1):
        print(f'\n📝 Test {idx}/{len(test_messages)}')
        print(f'💬 Message: "{message}"')
        print('🔄 Calling API...')
        
        try:
            start_time = time.time()
            
            response = requests.post(
                f'{server_url}/api/chatbot',
                json={
                    'message': message,
                    'patientResults': {
                        'eGFR': 75,
                        'status': 'Normal Function',
                        'riskLevel': 'Low',
                        'confidence': 95,
                        'heartRate': 72,
                        'temperature': 36.8
                    },
                    'conversationHistory': [],
                    'sessionId': f'test_session_{int(time.time())}'
                },
                headers={'Content-Type': 'application/json'},
                timeout=30
            )
            
            elapsed = time.time() - start_time
            
            if response.status_code == 200:
                data = response.json()
                if data.get('success'):
                    reply = data.get('reply', '')[:80]
                    print(f'✅ Success! Response: "{reply}..."')
                    print(f'⏱️ Response time: {elapsed:.2f}s')
                else:
                    print(f'⚠️ API returned false for success')
            else:
                print(f'❌ HTTP {response.status_code}')
                
        except requests.exceptions.Timeout:
            print(f'⏱️ Request timed out')
        except requests.exceptions.ConnectionError:
            print(f'❌ Connection error - server might have crashed!')
            return False
        except Exception as e:
            print(f'❌ Error: {e}')
            return False
        
        # Wait 2 seconds between tests
        time.sleep(2)
    
    print('\n' + '='*60)
    print('✅ ALL TESTS PASSED - Server is stable!')
    print('='*60 + '\n')
    return True

if __name__ == '__main__':
    try:
        success = test_chatbot()
        exit(0 if success else 1)
    except Exception as e:
        print(f'Fatal error: {e}')
        exit(1)

File: tests/test-chatbot-stability.js
// Extended chatbot test - multiple messages to test stability
require('dotenv').config();
const fetch = require('node-fetch');

async function testChatbotStability() {
    const serverURL = 'http://localhost:3000';
    
    console.log('\n═══════════════════════════════════════════════════════');
    console.log('🤖 EXTENDED CHATBOT STABILITY TEST');
    console.log('═══════════════════════════════════════════════════════\n');
    
    const testMessages = [
        'Hello! How are you?',
        'How do I keep my kidneys healthy?',
        'What is a kidney stone?',
        'What should I eat for kidney health?',
        'Is my kidney function normal at eGFR 75?',
        'What exercises are good for kidney patients?'
    ];
    
    for (let i = 0; i < testMessages.length; i++) {
        const message = testMessages[i];
        const testNum = i + 1;
        
        console.log(`\n📝 Test ${testNum}/${testMessages.length}`);
        console.log(`💬 User: "${message}"`);
        console.log('🔄 Waiting for response...');
        
        try {
            const response = await fetch(`${serverURL}/api/chatbot`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    message: message,
                    patientResults: {
                        eGFR: 75,
                        status: 'Normal Function',
                        riskLevel: 'Low',
                        confidence: 95,
                        heartRate: 72,
                        temperature: 36.8
                    },
                    conversationHistory: [],
                    sessionId: 'test_session_' + Date.now()
                })
            });
            
            if (!response.ok) {
                console.log(`❌ API Error: ${response.status}`);
                continue;
            }
            
            const data = await response.json();
            
            if (data.success) {
                const replyPreview = data.reply.substring(0, 100) + (data.reply.length > 100 ? '...' : '');
                console.log(`✅ Response received: "${replyPreview}"`);
            } else {
                console.log(`❌ API returned false for success`);
            }
        } catch (error) {
            console.log(`❌ Error: ${error.message}`);
        }
        
        // Wait 2 seconds between requests
        await new Promise(r => setTimeout(r, 2000));
    }
    
    console.log('\n═══════════════════════════════════════════════════════');
    console.log('✅ STABILITY TEST COMPLETE');
    console.log('If all tests passed, server is stable and won\'t crash!');
    console.log('═══════════════════════════════════════════════════════\n');
}

testChatbotStability();
